pkg/ipmi/ipmi.go

```diff
@@ -135,13 +135,13 @@ func (e *Event) marshall() ([]byte, error) {
 	}
 
 	// OEM timestamped
-	if buf.Bytes()[2] >= 0xC0 && buf.Bytes()[2] <= 0xDF {
+	if buf.Bytes()[2] >= 0xC0 {
```

Comment:
all this code is blindly referencing  an array, assuming it is long enough. I know that's not on you by maybe it's time to fix it anyway.

Reply:
You are right, I will add appropriate bound checks.